src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Jobs/WaitAzureRmRecoveryServicesBackupJob.cs

```diff
@@ -139,7 +140,12 @@ public override void ExecuteCmdlet()
                     }
 
                     // sleep for 30 seconds before checking again
-                    TestMockSupport.Delay(30 * 1000);
+                    string testMode = Environment.GetEnvironmentVariable("AZURE_TEST_MODE");
```

Comment:
@isra-fel  added this check to skip waiting in case Playback mode. seems like tests are still stuck. can you pls let me know the environment variable for /azp and the value it takes.

Reply:
Sorry I was wrong. We didn't set a envrionment when running tests. However there's a static field [TestMockSupport.RunningMocked](https://github.com/Azure/azure-powershell-common/blob/master/src/Common/TestMockSupport.cs#L24)  you can check.

In fact, I don't know why you change all the ` TestMockSupport.Delay()` , that should work perfectly fine here

Reply:
Besides, not only here, but all the "System.Threading.Sleep()` should be wrapped in an `If` clause, otherwise the tests will still time out

Reply:
@isra-fel :  included "TestMockSupport.RunningMocked" and skipped two failed tests for now. I changed  TestMockSupport.Delay() to Thread.Sleep() because while running the tests in record mode this was not putting a delay and was flooding our service with many calls. However it was working fine while running the cmdlets.